test/command.js

```diff
@@ -1402,4 +1402,17 @@ describe('Command', () => {
       }).to.throw(/.*\.usage\(\) description must start with \$0.*/)
     })
   })
+
+  // addresses https://github.com/yargs/yargs/issues/510
+  it('fails when the promise returned by the command handler rejects', (done) => {
+    const error = new Error()
+    yargs('foo')
```

Comment:
I think it might be worth adding a test for the happy path too (the promise resolving).

Reply:
Sure, but I can't think of anything to assert in that case... This feature does nothing when the handler result is thenable and it resolves.

I had actually fixed (?) my resolution handler in `lib/command.js` from a `() => {}` noop to `null` so that it would not show as an untested code path in the test coverage reports.

Anyone have a suggestion of what to assert? What do we need to watch out for in this case?

Reply:
hey, made comment above with recommendation about assertion; I suppose now that the fail handler is called async, you can just call `done()` here after your promise resolves?